packages/govtnz-ds-website/src/pages/prototypes/index.tsx

```diff
@@ -0,0 +1,309 @@
+        const { errorMessage, emailFormatError } = this.state[currentVal.name];
+
+        acc[currentVal.name] = {
+          value,
+          valid,
+          typeMismatch,
+          errorMessage,
+          emailFormatError,
+        };
+
+        return acc;
+      }, {});
+
+    return formfieldValues;
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+
+    const form = event.target;
+    const formValues = this.getformFieldvalues(form.elements);
+
+    const isFieldsValid = !Object.keys(formValues)
+      .map((value) => formValues[value])
+      .some((field) => !field.valid);
+
+    this.setState({ ...formValues, isFieldsValid });
+  };
+
+  render() {
+    const { email, username, textarea, radio, isFieldsValid } = this.state;
+    const successMessage = isFieldsValid ? 'block' : 'none';
+
+    return (
+      <>
+        <form
+          style={{ padding: '20px' }}
+          onSubmit={this.handleSubmit}
+          noValidate
+        >
+          <H1 styleSize="xlarge" id="main-heading">
+            Contact form
+          </H1>
+          <div
+            className={
+              username.valid
+                ? 'g-inputBlock-form-group'
+                : 'g-inputBlock-form-group--error'
+            }
+          >
+            {username.valid ? (
+              ''
+            ) : (
+              <ErrorMessage message={username.errorMessage} />
+            )}
+            <InputBlock
+              width="30"
+              maxLength={30}
+              label="What is your name?"
+              id="username"
+              name="username"
+              type="text"
+              autoComplete="autoComplete"
+              required
+            />
+          </div>
+          <div
+            className={
+              email.valid
+                ? 'g-inputBlock-form-group'
+                : 'g-inputBlock-form-group--error'
+            }
+          >
+            {email.valid ? (
+              ''
+            ) : (
+              <ErrorMessage
+                message={
+                  email.typeMismatch
+                    ? email.emailFormatError
+                    : email.errorMessage
+                }
+              />
+            )}
+
+            <InputBlock
+              width="30"
+              maxLength={30}
+              hintId="anyHintId3"
+              label="What is your email?"
+              autoComplete="autoComplete"
+              type="email"
+              name="email"
+              required
+            />
+          </div>
+          <div
+            className={
+              textarea.valid
+                ? 'g-inputBlock-form-group'
+                : 'g-textareaBlock-form-group--error'
+            }
+          >
+            {textarea.valid ? (
+              ''
+            ) : (
+              <ErrorMessage message={textarea.errorMessage} />
+            )}
+
+            <TextareaBlock
+              autoComplete="autoComplete"
```

Comment:
@Juliet-Brown Do we need an autoComplete value? Usually that's for prefilling fields that the browser might know about like address/email/phonenumber but I can't imagine that they could autocomplete a message. Also I don't think `autoComplete="autoComplete"` is a valid value.

Reply:
Yeah it probably doesn't It was throwing a type error about it when I removed it. I'll look at it a bit more in depth. thanks :)